lib/cartopy/io/img_tiles.py

```diff
@@ -48,22 +49,39 @@ class GoogleWTS(six.with_metaclass(ABCMeta, object)):
     A "tile" in this class refers to the coordinates (x, y, z).
 
     """
+    _MAX_THREADS = 24
+
     def __init__(self, desired_tile_form='RGB'):
         self.imgs = []
         self.crs = ccrs.Mercator.GOOGLE
         self.desired_tile_form = desired_tile_form
 
     def image_for_domain(self, target_domain, target_z):
         tiles = []
-        for tile in self.find_images(target_domain, target_z):
+
+        def fetch_tile(tile):
             try:
                 img, extent, origin = self.get_image(tile)
             except IOError:
-                continue
+                # Some services 404 for tiles that aren't supposed to be
+                # there (e.g. out of range).
+                raise
```

Comment:
Should you even bother with catching this, then?

Reply:
I had it there as a form of documentation so that it is clear that this method can and does raise. I have no major issue removing it, but equally I quite like the fact that you can see very quickly that this thing can raise an IOError (and that the consumer should deal with this).

Your call on this.